controllers/storagecluster/external_resources.go

```diff
@@ -552,7 +523,7 @@ func (r *StorageClusterReconciler) createExternalModeStorageClasses(sccs []Stora
 			if existing.DeletionTimestamp != nil {
 				return fmt.Errorf("failed to restore StorageClass  %s because it is marked for deletion", existing.Name)
 			}
-			if !reflect.DeepEqual(sc.Parameters, existing.Parameters) {
+			if !reflect.DeepEqual(sc.Parameters, existing.Parameters) || existing.Labels[util.ExternalClassLabelKey] == "" {
```

Comment:
Why are you checking for an empty value? What if it was not an empty value in the prev reconcile?
I think you need to check the diff in value between the actual state (existing) and desired state (sc)